libs/wingsdk/src/target-tf-aws/api.ts

```diff
@@ -0,0 +1,136 @@
+/* eslint-disable import/order */
+import { join } from "path";
+import { ApiGatewayMethod } from "@cdktf/provider-aws/lib/api-gateway-method";
+import { ApiGatewayResource } from "@cdktf/provider-aws/lib/api-gateway-resource";
+import { ApiGatewayRestApi } from "@cdktf/provider-aws/lib/api-gateway-rest-api";
+import { LambdaPermission } from "@cdktf/provider-aws/lib/lambda-permission";
+import { DataAwsRegion } from "@cdktf/provider-aws/lib/data-aws-region";
+import { DataAwsCallerIdentity } from "@cdktf/provider-aws/lib/data-aws-caller-identity";
+
+import { Construct } from "constructs";
+import { core } from "..";
+import * as cloud from "../cloud";
+import { convertBetweenHandlers } from "../convert";
+import { Code } from "../core/inflight";
+import { Function } from "./function";
+import { ApiGatewayIntegration } from "@cdktf/provider-aws/lib/api-gateway-integration";
+import { IResource } from "../core";
+
+/**
+ * AWS Implementation of `cloud.Api`.
+ */
+export class Api extends cloud.ApiBase {
```

Comment:
I get your point, but do you have an idea how the OpenAPI spec can be generated? The only advantage I can see right now, is that I don't have to build the ApiRessources from the api path. 

If I take a look at the aws docs https://docs.aws.amazon.com/de_de/apigateway/latest/developerguide/api-gateway-request-validation-sample-api-swagger.html
the openapi spec needs to be extended with amazon specific extensions. Even if the general spec with its routes and methods would be independent, the openapi extensions would be vendor specific.

Reply:
maybe a library like this one can help generating the openapi spec
https://github.com/metadevpro/openapi3-ts

Reply:
I like the idea...

Maybe in terms of implementation we could take a look at how CDK generates APIs from specs, I think they support a feature like this here: https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_apigateway.SpecRestApi.html

BTW, there might be a chance the `spec` will need to be a lazy property since `post()` / `get()` / `add_route()` would modify the spec. We don't have something like this in the SDK yet because we haven't had a use case yet, but here is a test showing how they work: https://github.com/cdk8s-team/cdk8s-core/blob/2.x/test/tokens.test.ts

Reply:
Yes good point about the lazy property. Alternatively I was thinking we can offer "preSynth" hooks on any construct in the Wing SDK. 

We had this at some point in the CDK and removed it because it's  quite abusable, but I think there's a use case for it.

It's also aligned with @hasanaburayyan's work on plugins.

Reply:
Got it. Maybe we can discuss this in a separate issue -- my only concern with respect to synth hooks is that they don't always play well with escape hatches.

(Concretely, I could imagine someone creating a wing library where an AwsIamRole is created during presynth (using terraform-cdk). Then, when I want to make a library that composes over their library, if I want to do something like create a resource per Iam statement or if I want to apply CDKTF escape hatches, all my library logic will have to be in presynth, and it seems like we're back at square one where someone wants a postPreSynth stage etc.)

Reply:
In any case I'll leave it up to @RaphaelManke which direction you'd like to choose, the OpenAPI direction sounds promising but we can also continue with the direction that does not use OpenAPI to start and maybe refactor later. 🙂

Reply:
for now I added a `addResource` helper function that adds a route to the api. I still like the openapi spec idea because it would also enable the wing console to render the api spec and let the user invoke the api there. But I am not very familiar with the Token resolution and the lazy method.

Reply:
Sorry for the back and forth, I now refactored to use the openApi approach and generate a spec file and the generate the restApi from that. This makes the generation actually very simple because one does not have to create resource, method and integration as terraform resources. On top of that it fixes the deployment issue. Not a deleted resource is not deployed anymore. I will continue to follow the the openapi spec approach.